lucene/highlighter/src/test/org/apache/lucene/search/uhighlight/TestUnifiedHighlighter.java

```diff
@@ -959,4 +961,84 @@ protected PassageFormatter getFormatter(String field) {
     ir.close();
   }
 
+  public void testBooleanWithSpanAndOverlappingTerms() throws IOException {
```

Comment:
I will fix the name of the test to clarify and move to the appropriate file. Regarding PhraseHelper, that should probably be a separate issue/commit I'd think, no? Agree it should be have package visibility rather than be public.